automerge-backend/src/backend.rs

```diff
@@ -84,12 +82,12 @@ impl Backend {
 
     fn apply(
         &mut self,
-        mut changes: Vec<Change>,
+        changes: Vec<Change>,
         actor: Option<(amp::ActorId, u64)>,
     ) -> Result<amp::Patch, AutomergeError> {
         let mut pending_diffs = HashMap::new();
 
-        for change in changes.drain(..) {
+        for change in changes.into_iter() {
```

Comment:
Here using `drain` would have made a lot of sense if you had a mutable reference to the vector, and you expected the caller to still use the empty one after the changes had been applied.